CharlieBackend.Business/Services/FileImportService.cs

```diff
@@ -0,0 +1,155 @@
+                        StartDate = fileLine.StartDate,
+                        FinishDate = fileLine.FinishDate,
+                    };
+
+                    importedGroups.Add(fileLine);
+                    _unitOfWork.StudentGroupRepository.Add(group);
+                    rowCounter++;
+                }
+                catch (FormatException ex)
+                {
+                    _unitOfWork.Rollback();
+                    return Result<List<StudentGroupFileModel>>.Error(ErrorCode.ValidationError,
+                        "The format of the inputed data is incorrect.\n" + ex.Message);
+                }
+                catch (DbUpdateException ex)
+                {
+                    _unitOfWork.Rollback();
+                    return Result<List<StudentGroupFileModel>>.Error(ErrorCode.ValidationError,
+                        "Inputed data is incorrect.\n" + ex.Message);
+                }
+            }
+
+            await _unitOfWork.CommitAsync();
+            return Result<List<StudentGroupFileModel>>
+                .Success(_mapper.Map<List<StudentGroupFileModel>>(importedGroups));
+        }
+
+        private async Task IsValueValid(StudentGroupFileModel fileLine, int rowCounter)
+        {
+            List<long> existingCourseIds = new List<long>();
+            List<string> existingGroupNames = new List<string>();
+
+            foreach (Course course in await _unitOfWork.CourseRepository.GetAllAsync())
+            {
+                existingCourseIds.Add(course.Id);
+            }
+
+            foreach (StudentGroup group in await _unitOfWork.StudentGroupRepository.GetAllAsync())
+            {
+                existingGroupNames.Add(group.Name);
+            }
+
+
+            if (fileLine.CourseId.Replace(" ", "") == "")
+            {
+                throw new FormatException("CourseId field shouldn't be empty.\n" +
+                    $"Problem was occured in col B, row {rowCounter}");
+            }
+
+            if (fileLine.Name == "")
+            {
+                throw new FormatException("Name field shouldn't be empty.\n" +
+                    $"Problem was occured in col C, row {rowCounter}");
+            }
+
+            if (fileLine.StartDate > fileLine.FinishDate)
+            {
+                throw new FormatException("StartDate must be less than FinishDate.\n" +
+                    $"Problem was occured in col D/E, row {rowCounter}.");
+            }
+
+            if (!existingCourseIds.Contains(Convert.ToInt64(fileLine.CourseId)))
+            {
+                throw new DbUpdateException($"Course with id {fileLine.CourseId} doesn't exist.\n" +
+                   $"Problem was occured in col B, row {rowCounter}.");
+            }
+
+            if (existingGroupNames.Contains(fileLine.Name))
+            {
+                throw new DbUpdateException($"Group with name {fileLine.Name} already exist.\n" +
+                   $"Problem was occured in col C, row {rowCounter}.");
```

Comment:
Why are you just throw exceptions instead of using Result<t>?

Reply:
When you throw exception like this you are make fall whole program

Reply:
than I'll catch it and returns Results

Reply:
this code is in the method that shouldn't returns Results